### PR TITLE
Add isTransparent prop to section header / toggle section header

### LINKED ISSUE #1684: Add section header variant that has a transparent background
# Feature Request

## Description
In our UX callouts, there is a section header variation that includes a transparent background. When the section header component was originally coded, it looks like didn't include this feature. This work would involve adding a prop to enable this look for the section header.

## Additional Context / Screenshots
screen shot 2018-07-09 at 8 46 28 am

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Resolves #1684 

### Deployment Link
https://terra-core-deployed-pr-1708.herokuapp.com/#/components/terra-section-header/section-header/section-header

### CODE REVIEW

File: packages/terra-toggle-section-header/tests/wdio/toggle-section-header-spec.js [hunk 1]
```diff
@@ -14,4 +14,11 @@ describe('ToggleSectionHeader', () => {
     Terra.should.beAccessible();
     Terra.should.matchScreenshot();
   });
+
+  describe('Transparent', () => {
+    before(() => browser.url('/#/raw/tests/terra-toggle-section-header/toggle-section-header/transparent-toggle-section-header'));
+
+    Terra.should.beAccessible();
+    Terra.should.matchScreenshot();
+  });
```
Comment (Blackop778) [inline packages/terra-toggle-section-header/tests/wdio/toggle-section-header-spec.js]: Should we be testing that the section header can be toggled?

**Comment by mmalaker** [comment]
Functional verification completed. The section header examples now include the transparent section header.

File: packages/terra-section-header/src/SectionHeader.module.scss [hunk 2]
```diff
@@ -52,6 +52,45 @@
...
+        border-bottom: var(--terra-transparent-section-header-hover-border-bottom, 0);
+        border-left: var(--terra-transparent-section-header-hover-border-left, 0);
+        border-right: var(--terra-transparent-section-header-hover-border-right, 0);
+        border-top: var(--terra-transparent-section-header-hover-border-top, 0);
+        box-shadow: var(--terra-transparent-section-header-hover-box-shadow);
+        color: var(--terra-transparent-section-header-hover-color, #1c1f21);
+      }
+
+      &:focus {
+        background: var(--terra-transparent-section-header-focus-background, rgba(89, 159, 253, 0.3));
```
Comment (neilpfeiffer) [inline packages/terra-section-header/src/SectionHeader.module.scss]: focus:
`background: var(--terra-transparent-section-header-active-background, rgba(0, 0, 0, 0.07));`

File: packages/terra-section-header/src/SectionHeader.module.scss [hunk 3]
```diff
@@ -52,6 +52,45 @@
...
+        border-right: var(--terra-transparent-section-header-focus-border-right, 0);
+        border-top: var(--terra-transparent-section-header-focus-border-top, 0);
+        box-shadow: var(--terra-transparent-section-header-focus-box-shadow, 0 0 1px 3px rgba(76, 178, 233, 0.5), 0 0 7px 4px rgba(76, 178, 233, 0.35));
+        color: var(--terra-transparent-section-header-focus-color, #1c1f21);
+        outline: none;
+      }
+
+      &:active,
+      &.is-active {
+        background: var(--terra-transparent-section-header-active-background, rgba(89, 159, 253, 0.3));
```
Comment (neilpfeiffer) [inline packages/terra-section-header/src/SectionHeader.module.scss]: active:
`background: var(--terra-transparent-section-header-active-background, rgba(0, 0, 0, 0.07));`

**Comment by neilpfeiffer** [comment]
Tested: PR #1708
Resolves: Issue #1684
 
**Reviewed:**
- initial release of new prop needed for theming, will update visual details in future as needed.
 
\
\
Verified by @neilpfeiffer, Ready for Release.
 
 
**Pass** |
-----------|
UX Review|

**Pull request merged**